src/utils/playerHasWon.js

```diff
@@ -0,0 +1,48 @@
+import { checkWin } from './checkWin';
+import { isBoardFull } from './isBoardFull';
+
+export function playerHasWon(gameState, player) {
+  if (isBoardFull(gameState)) {
+    return false; // Draw
+  }
+
+  const winCount = gameState.winCondition[0];
+
+  // Find a vertical sequence for player
+  for (let col = 0; col < gameState.boardHeights.length; col++) {
```

Comment:
I think without doing the optimization we can simplify all of this code back down to a single loop and just specifying the "directions". You had the key insight that you only need 4 directions instead of 8.

So we would have:

for each col
  for each row
    for each direction (0, 1), (1, 0), (1, 1) (1, -1)
      checkWin